source/css/sass/_mixins--spacing.scss

```diff
@@ -1,44 +1,188 @@
 @import "utility-functions";
-@import "variables--baseline-grid";
-@import "variables--breakpoints";
+@import "variables";
 
+// NOTE: mixins beginning with underscore "_" are internal to this file only: they MUST NOT be used externally.
+
+@mixin _padding-left($size-in-px) {
+  @if $size-in-px == 0 {
+    padding-left: 0;
+  } @else {
+    padding-left: #{$size_in_px}px;
+    padding-left: #{get-rem-from-px($size_in_px)}rem;
+  }
+}
+
+@mixin _padding-right($size-in-px) {
+  @if $size-in-px == 0 {
+    padding-right: 0;
+  } @else {
+    padding-right: #{$size_in_px}px;
+    padding-right: #{get-rem-from-px($size_in_px)}rem;
+  }
+}
+
+// Fallbacks for CSS logical properties contained within this mixin require the following treatment of HTML dir attributes:
+//  - document level: always specified, via the HTML element
+//  - block level: specified on every element within a block describing a direction switch.
+//
+// For example:
+// <html lang="..." dir="ltr">
+// ...
+// <div>
+//   Doesn't need a dir attribute. Most cases will be like this.
+// </div>
+//
+//<div class="test" dir="rtl">
+//
+//  This block changes the text direction. Every descendant element must have its own dir attribute....
+//
+//  <div class="test" dir="rtl">... even if the direction doesn't change.</div>
+//
+//  <div class="test" dir="ltr">But obviously also when it does.</div>
+//
+//</div>
 @mixin padding($size_in_px, $dimension: "") {
-  @if $dimension == top or $dimension == right or $dimension == bottom or $dimension == left {
+  @if $dimension == "" {
     @if $size_in_px == 0 {
-      padding-#{$dimension}: 0;
+      padding: 0;
     } @else {
-      padding-#{$dimension}: #{$size_in_px}px;
-      padding-#{$dimension}: #{get-rem-from-px($size_in_px)}rem;
+      padding: #{$size_in_px}px;
+      padding: #{get-rem-from-px($size_in_px)}rem;
     }
   }
-  @else if $dimension == "" and type_of($size_in_px) == list {
-    $parts-px: ();
-    $parts-rem: ();
-    @each $size in $size_in_px {
-      @if $size == 0 {
-        $parts-px: append($parts-px, 0, "space");
+  @if $dimension == "inline" {
+    @if $size_in_px == 0 {
+      padding-left: 0;
+      padding-right: 0;
+      padding-inline: 0;
+    } @else if type_of($size_in_px) == list and length($size_in_px) >= 2 {
+
+      $firstValue: nth($size_in_px, 1);
+      $secondValue: nth($size_in_px, 2);
+
+      @if $firstValue == $secondValue {
+
+        @include _padding-left($firstValue);
+        @include _padding-right($secondValue);
+
       } @else {
-        $parts-px: append($parts-px, #{$size}px, "space");
+
+        html[dir="ltr"] &:not([dir]),
+        &[dir="ltr"] {
```

Comment:
Is there a way to avoid duplicating the 'is LTR' and 'is RTL' logic?

Reply:
Maybe. I'll take a look.

Reply:
Yup, got something that will work using `@contents`. Commit in prep.